src/mesh/mesh_adaptation.cpp

```diff
@@ -0,0 +1,117 @@
+#include "mesh_adaptation.h"
+
+namespace PHiLiP {
+
+template <int dim, typename real, typename MeshType>
+MeshAdaptation<dim,real,MeshType>::MeshAdaptation(double critical_res_input, int total_ref_cycle, double refine_frac, double coarsen_frac)
+    : critical_residual(critical_res_input)
+    , total_refinement_cycles(total_ref_cycle)
+    , refinement_fraction(refine_frac)
+    , coarsening_fraction(coarsen_frac)
+    , current_refinement_cycle(0)
+    , pcout(std::cout, dealii::Utilities::MPI::this_mpi_process(MPI_COMM_WORLD)==0)
+    {}
+
+template <int dim, typename real, typename MeshType>
+void MeshAdaptation<dim,real,MeshType>::adapt_mesh(std::shared_ptr< DGBase<dim, real, MeshType> > dg)
+{
+    if (!refine_mesh)
+    {
+        return;
+    }
+
+    if (total_refinement_cycles == current_refinement_cycle)
+    {
+        return;
+    }
+
+    cellwise_errors.reinit(0);
+    cellwise_errors.reinit(dg->high_order_grid->triangulation->n_active_cells());
+    compute_cellwise_errors(dg);
+
+    fixed_fraction_isotropic_refinement_and_coarsening(dg);
+    current_refinement_cycle++;
+    pcout<<"Refined"<<std::endl;
+
+    return;
+}
+
+template <int dim, typename real, typename MeshType>
+void MeshAdaptation<dim,real,MeshType>::compute_cellwise_errors(std::shared_ptr< DGBase<dim, real, MeshType> > dg)
+{
+    compute_max_cellwise_residuals(dg); // Future extension: Error depends on parameters input (i.e. this function computes residual or goal-oriented error).
+    return;
+}
+
+
+template <int dim, typename real, typename MeshType>
+void MeshAdaptation<dim,real,MeshType>::compute_max_cellwise_residuals(std::shared_ptr< DGBase<dim, real, MeshType> > dg)
+{
+    std::vector<dealii::types::global_dof_index> dofs_indices;
+    for (const auto &cell : dg->dof_handler.active_cell_iterators()) 
+    {
+         if (!cell->is_locally_owned()) 
+         continue;
+ 
+         const int i_fele = cell->active_fe_index();
+         const dealii::FESystem<dim,dim> &fe_ref = dg->fe_collection[i_fele];
+         const unsigned int n_dofs_cell = fe_ref.n_dofs_per_cell();
+         dofs_indices.resize(n_dofs_cell);
+         cell->get_dof_indices (dofs_indices);
+         double max_residual = 0;
+         for (unsigned int idof = 0; idof < n_dofs_cell; ++idof) 
+         {
+            const unsigned int index = dofs_indices[idof];
+            const double res = std::abs(dg->right_hand_side[index]);
+            if (res > max_residual) 
+                max_residual = res;
+         }
+         cellwise_errors[cell->active_cell_index()] = max_residual;
+     }
+
+    return;
```

Comment:
Unnecessary "return;" at the end of a void function. Same applies for some of the other functions you provided.

Reply:
Removed return at the end.